velox/exec/fuzzer/WriterFuzzer.cpp

```diff
@@ -579,20 +687,64 @@ std::map<std::string, int32_t> WriterFuzzer::getPartitionNameAndFilecount(
 RowTypePtr WriterFuzzer::generateOutputType(
     const std::vector<std::string>& names,
     const std::vector<TypePtr>& types,
-    const int32_t partitionCount,
     const int32_t bucketCount) {
   std::vector<std::string> outputNames{names};
   std::vector<TypePtr> outputTypes;
   for (auto type : types) {
     outputTypes.emplace_back(type);
   }
-  if (isBucketed(partitionCount, bucketCount)) {
+  if (bucketCount > 0) {
     outputNames.emplace_back("$bucket");
     outputTypes.emplace_back(INTEGER());
   }
 
   return {ROW(std::move(outputNames), std::move(outputTypes))};
 }
 
+std::string WriterFuzzer::sortSql(
+    const std::shared_ptr<HiveConnectorSplit>& split,
+    const std::vector<std::string>& names,
+    const std::vector<TypePtr>& types,
+    int32_t partitionOffset,
+    const std::vector<std::string>& partitionKeys,
+    const std::vector<std::shared_ptr<const HiveSortingColumn>>& sortBy) {
+  // For a split, extract the partition filters and bucket filters.
```

Comment:
Can we parse the split info and then pass to reference DB? So we don't need to expose this split directly to reference DB? thanks!